process/economics/economicsData.go

```diff
@@ -580,6 +591,16 @@ func (ed *economicsData) ComputeGasUsedAndFeeBasedOnRefundValue(tx data.Transact
 	return ed.ComputeGasUsedAndFeeBasedOnRefundValueInEpoch(tx, refundValue, currentEpoch)
 }
 
+// ComputeGasUnitsFromRefundValue will compute the gas unit based on the refund value
+func (ed *economicsData) ComputeGasUnitsFromRefundValue(tx data.TransactionWithFeeHandler, refundValue *big.Int) uint64 {
+	currentEpoch := ed.enableEpochsHandler.GetCurrentEpoch()
```

Comment:
epoch should be the one of tx here as well, so perhaps get it as parameter

Reply:
changed